src/Lachain.Storage/State/StateManager.cs

```diff
@@ -143,13 +145,20 @@ public void Rollback()
 
         public void Commit()
         {
-            _balanceManager.Commit();
-            _contractManager.Commit();
-            _storageManager.Commit();
-            _transactionManager.Commit();
-            _blockManager.Commit();
-            _eventManager.Commit();
-            _validatorManager.Commit();
+            var batch = new RocksDbAtomicWrite(_dbContext);
+            _balanceManager.Commit(batch);
+            _contractManager.Commit(batch);
+            _storageManager.Commit(batch);
+            _transactionManager.Commit(batch);
+            _blockManager.Commit(batch);
+            _eventManager.Commit(batch);
+            _validatorManager.Commit(batch);
+            batch.Commit();
+        }
+
+        public void Commit(RocksDbAtomicWrite batch)
```

Comment:
why do we need this ?

Reply:
StateManager inherits from IStateManager and IStateManager inherits from ISnapshotManager<IBlockchainSnapshot>, now SnapshotManager also inherits from ISnapshotManager. I needed to put Commit(batch) in the ISnapshotManager and as a result I also had to put it in StateManager. This looks really stupid but I can't think of any other way without massive change in the architecture.

Reply:
maybe we can add some comments or throw exception inside this method to make sure none uses it accidentally.